.github/workflows/publish.yml

```diff
@@ -0,0 +1,32 @@
+name: Publish to Hex
+
+on:
+  push
+
+env:
+  HEX_API_KEY: ${{ secrets.HEX_API_KEY }}
```

Comment:
@AnasNaouchi Can we scope this env to only the publish action?

Reply:
Although the only job in this workflow is publish so publish will be the sole user of the env we can also set the env at job/step level

Reply:
Sorry, I think we can scope even further to the actual `run: mix hex.publish --dry-run` step.

Reply:
Shall switch now to prod ?